### DISCUSSION TITLE
Slider - changing/extending range.

### DISCUSSION THREAD

**Discussion by Latawiec** [ArthurSonzogni/FTXUI#597] [Q&A]
Hello,
I wanted to extend the range of a Slider. Slider is in a Renderer() that wraps it in a hbox.
How I wanted to achieve this is by swapping contents of the shared_ptr, but if I do that events don't go through it seems.
```
Component slider = Slider("", 1, 1, 10, 1);
auto mainComp = Renderer(slider, [&]{
   return hbox({slider->Render()}); 
};
// ... later in the code...
Slider("", 1, 1, 20, 1).swap(slider );
```

From what I see the Renderer will just store any children I pass to it, how come it doesn't just keep going if I swap the content from under the shared_ptr?

How else can I achieve this?

**Comment by Latawiec** [comment]
Nevermind, clearly the shared_ptr I passed won't be updated. 

So question remaining is - is there any other way I can solve this?

**Reply by Latawiec** [reply]
Ok I got it:
```
auto parent = slider->Parent();
slider->Detach();
slider = Slider(...); // new setup
parent->Add(slider);
```

I'm sorry for the mess I've made in this Discussion. I will now shut up.

**Reply by ArthurSonzogni** [reply]
Hello!

You can also bind most argument by reference instead of by value:
```cpp
// The data:
int value = 0;
int min = 1;
int max = 10;

// The UI:
Component slider = Slider("", &value, &min, &max, 1);
auto mainComp = Renderer(slider, [&]{
   return hbox({slider->Render()}); 
};

// ... later in the code... update the data.
max = 100;
```

**Reply by Latawiec** [reply]
Wow sick! I should have checked the constructor signature. Thank you
